bundles/org.openhab.binding.smarther/src/main/java/org/openhab/binding/smarther/internal/handler/SmartherBridgeHandler.java

```diff
@@ -0,0 +1,699 @@
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerService;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.smarther.internal.account.SmartherAccountHandler;
+import org.openhab.binding.smarther.internal.account.SmartherNotificationHandler;
+import org.openhab.binding.smarther.internal.api.SmartherApi;
+import org.openhab.binding.smarther.internal.api.dto.Location;
+import org.openhab.binding.smarther.internal.api.dto.Module;
+import org.openhab.binding.smarther.internal.api.dto.ModuleStatus;
+import org.openhab.binding.smarther.internal.api.dto.Notification;
+import org.openhab.binding.smarther.internal.api.dto.Plant;
+import org.openhab.binding.smarther.internal.api.dto.Program;
+import org.openhab.binding.smarther.internal.api.dto.Sender;
+import org.openhab.binding.smarther.internal.api.dto.Subscription;
+import org.openhab.binding.smarther.internal.api.exception.SmartherAuthorizationException;
+import org.openhab.binding.smarther.internal.api.exception.SmartherGatewayException;
+import org.openhab.binding.smarther.internal.config.SmartherBridgeConfiguration;
+import org.openhab.binding.smarther.internal.discovery.SmartherModuleDiscoveryService;
+import org.openhab.binding.smarther.internal.model.BridgeStatus;
+import org.openhab.binding.smarther.internal.model.ModuleSettings;
+import org.openhab.binding.smarther.internal.util.StringUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@code SmartherBridgeHandler} class is responsible of the handling of a Smarther Bridge thing.
+ * The Smarther Bridge is used to manage a set of Smarther Chronothermostat Modules registered under the same
+ * Legrand/Bticino account credentials.
+ *
+ * @author Fabio Possieri - Initial contribution
+ */
+@NonNullByDefault
+public class SmartherBridgeHandler extends BaseBridgeHandler
+        implements SmartherAccountHandler, SmartherNotificationHandler, AccessTokenRefreshListener {
+
+    private static final long POLL_INITIAL_DELAY = 5;
+
+    private final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    private final OAuthFactory oAuthFactory;
+    private final HttpClient httpClient;
+
+    // Field members assigned in initialize method
+    private @NonNullByDefault({}) Future<?> pollFuture;
+    private @NonNullByDefault({}) OAuthClientService oAuthService;
+    private @NonNullByDefault({}) SmartherApi smartherApi;
+    private @NonNullByDefault({}) SmartherBridgeConfiguration config;
+
+    // Bridge local status
+    private @NonNullByDefault({}) ExpiringCache<List<Location>> locationCache;
+    private @NonNullByDefault({}) BridgeStatus bridgeStatus;
```

Comment:
Please avoid using `@NonNullByDefault({})`. Use `@Nullable` instead. Most of the null checker warnings can be resolved by caching the fields to local variables and performing your logic on those local variables instead.

Reply:
Thanks, same suggested changes I've applied them to `SmartherModuleHandler` as well.